packages/structured-clone/build.js

```diff
@@ -19,6 +19,7 @@ import { isCjsPackage } from "../../scripts/esbuild-module-check.mjs";
         format: "esm",
         target: "node14",
         external: [...explicitDeps],
+        conditions: ["_bundle"],
```

Comment:
I don't like the idea of adding a custom condition. Why is this needed?

Reply:
It is a key addition of the PR that makes the build work without using Typescript project references.

TS project reference allows Typescript to resolve import statements that import from another workspace package, such as `@unified-latex/unified-latex-types`, to their respective `.ts` source files or `.d.ts` declaration files depending on circumstances (more on it in the original post). 

If TS project reference is removed, Typescript and ESBuild will follow the native nodejs module resolution algorithm, and it involves finding matching `exports` condition, which will be `import` if without `_bundle`, and it will then look for `/dist/index.js` file, which may or may not exist at that point. `_bundle` is there to point it to the `.ts` source files.

Reply:
Frankly I don't like this as well, it feels like this custom condition is not designed for a use-case like this. I'll try to consult how other large Typescript monorepos are doing. That said, it was definitely an improvement in terms of dev experience because there's no longer any surprising build errors.